lib/link.js

```diff
@@ -127,6 +127,7 @@ Link.prototype.detach = function() {
 Link.prototype.forceDetach = function() {
   debug('Force-detach for ' + this.name + '. Current state: ' + this.state());
   this.linkSM.forceDetach();
+  this._detached();
```

Comment:
I'm not quite sure, some extra tests around this would help a lot. The issue is in a basic test that a forceDetach comes in and the links are never cleaned up (because `_detached`) isn't called.